pxf/pxf-jdbc/src/main/java/org/apache/hawq/pxf/plugins/jdbc/JdbcPartitionFragmenter.java

```diff
@@ -111,30 +110,40 @@ public JdbcPartitionFragmenter(InputData inConf) throws UserDataException {
             partitionBy = inConf.getUserProperty("PARTITION_BY").split(":");
             partitionColumn = partitionBy[0];
             partitionType = PartitionType.getType(partitionBy[1]);
-        } catch (IllegalArgumentException e1) {
+        } catch (IllegalArgumentException | ArrayIndexOutOfBoundsException e1) {
             throw new UserDataException("The parameter 'PARTITION_BY' invalid, the pattern is 'column_name:date|int|enum'");
         }
 
+        //parse and validate parameter-RANGE
         try {
-            range = inConf.getUserProperty("RANGE").split(":");
+            if (inConf.getUserProperty("RANGE") != null) {
+                range = inConf.getUserProperty("RANGE").split(":");

```

Comment:
calling same function twice, would be better to call it once saving result and then using the result for further logic